app/views/users/show.html.erb

```diff
@@ -0,0 +1,83 @@
+<main class="container">
+    <!-- Header of the section -->
+    <div class="m-5 text-center">
+        <h1>Hi <%= @user.name %> </h1>
+
+        <p>Here are all your related events.</p>
+    </div>
+    <!-- Second header -->
+    <h2>Events you are hosting</h2>
+    <div class="row">
+
+        <div class="col-lg-6">
+            <div class="wrap">
+                <h4>Upcoming events</h4>
+                <% @user.events.upcoming(Time.now).each do |event| %>
+                <div class="events mb-4">
+                    <p><%= event.description %></p>
+                    <small class="float-right"><%= time_ago_in_words(event.date) %> from now </small>
+                </div>
+                <% end %>
+            </div>
+        </div>
+
+        <div class="col-lg-6">
+            <div class="wrap">
+                <h4>Past events</h4>
+                <% @user.events.past(Time.now).each do |event| %>
+                <div class="events mb-3">
+                    <p><%= event.description %> </p>
+                    <small class="float-right"><%= time_ago_in_words(event.date) %> ago</small>
+                </div>
+                <% end %>
+
+            </div>
+        </div>
+    </div>
+
+    <!-- third header -->
+    <h2> Events you are attending: </h2>
+
+    <div class="row">
+
+        <div class="col-lg-6">
+            <div class="wrap">
+                <h4>Past events</h4>
+
+                <% @user.attendances.not_invited.each do |attendance| %>
+                <% if @events.past(Time.now).exists?(attendance.event_id) %>
+                <div class="mb-bit events">
+                    <%= @events.past(Time.now).find(attendance.event_id).description %> <br>
+                    <%= @events.past(Time.now).find(attendance.event_id).date %> <br>
+
+                </div>
+
+                <%end%>
+
+                <% end %>
+            </div>
+        </div>
+        <div class="col-lg-6">
+            <div class="wrap">
+                <h4>Upcoming events</h4>
+
+                <% @user.attendances.not_invited.each do |attendance| %>
+                <% if @events.upcoming(Time.now).exists?(attendance.event_id) %>
+
+                <div class="events mb-bit">
+                    <%= @events.upcoming(Time.now).find(attendance.event_id).description %> <br>
```

Comment:
Guys, it is better if you move this query to the controller. Views shouldn't handle logic like queries.

Reply:
Also, the `Time.now` might be handled in the scope you created in the model, this applies for upcoming and past scopes.